src/rir.rs

```diff
@@ -121,17 +143,17 @@ impl FloatSinc for f64 {
     }
 }
 
-fn sim_microphone(p: &Position, a: &Angle, t: &MicrophoneType) -> f64 {
-    match t {
+fn sim_microphone(p: Position, recv: &Receiver) -> f64 {
+    match recv.microphone_type {
         MicrophoneType::Hypercardioid => 1.0,
         _ => {
-            let rho: f64 = t.clone().into();
-            let vartheta = (p.z / (p.x.powi(2) + p.y.powi(2) + p.z.powi(2)).sqrt()).acos();
-            let varphi = p.y.atan2(p.x);
+            let (vartheta, varphi) = ((p.z / p.radius()).acos(), p.y.atan2(p.x));
```

Comment:
that can be uniformed with the line below by adding a `fn angle` to a position, calculating angle for spherical coordinates

Reply:
you could also have a CartesianPoint and SphericalPoint structures and provide conversion functions for them